src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1412,6 +1412,50 @@ private boolean validateServerName(String nameInCert) throws CertificateExceptio
                 return false;
             }
 
+            int wildcardIndex = nameInCert.indexOf("*");
+
+            // Respect wildcard. If wildcardIndex is larger than -1, then we have a wildcard.
+            if (wildcardIndex >= 0) {
+                // We do not allow wildcards to exist past the first period.
+                if (wildcardIndex > nameInCert.indexOf(".")) {
+                    return false;
+                }
+                
+                // We do not allow wildcards in IDNs.
+                if (nameInCert.startsWith("xn--")) {
+                    return false;
+                }
+                
+                // We do not allow * plus a top-level domain.
+                // This if statement counts the number of .s in the nameInCert. If it's 1 or less, then reject it.
+                // This also catches cases where nameInCert is just *
+                if ((nameInCert.length() - nameInCert.replace(".", "").length()) <= 1) {
```

Comment:
This probably works fine but is a weird workaround to existing API. The following should work just as well and is more readable/intuitive.
```java
if (nameInCert.chars().filter(ch -> ch == '.').count() <= 1)
```

Reply:
i think this looks more confusing, so i'll leave it as is